pkg/hive/hive.go

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package hive
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"go.uber.org/fx"
+	"go.uber.org/fx/fxevent"
+	"go.uber.org/fx/fxtest"
+
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "hive")
+)
+
+const (
+	defaultStartTimeout = 5 * time.Minute
+	defaultStopTimeout  = time.Minute
+	defaultEnvPrefix    = "CILIUM_"
```

Comment:
What are these start and stop timeouts?

Reply:
They're the amount of time there is for the start and stop hooks to execute, e.g. it's the amount of time after which the context.Context passed to the hooks gets cancelled. I'll add a comment :)